pkg/testutils/kyma.go

```diff
@@ -0,0 +1,134 @@
+package testutils
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+	"github.com/kyma-project/lifecycle-manager/pkg/testutils/builder"
+	"github.com/kyma-project/lifecycle-manager/pkg/util"
+	"github.com/kyma-project/lifecycle-manager/pkg/watcher"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	ErrStatusModuleStateMismatch = errors.New("status.modules.state not match")
+	ErrKymaNotDeleted            = errors.New("kyma CR not deleted")
+)
+
+func NewTestKyma(name string) *v1beta2.Kyma {
+	return newKCPKymaWithNamespace(name, v1.NamespaceDefault, v1beta2.DefaultChannel, v1beta2.SyncStrategyLocalClient)
+}
+
+func NewKymaForE2E(name, namespace, channel string) *v1beta2.Kyma {
+	kyma := newKCPKymaWithNamespace(name, namespace, channel, v1beta2.SyncStrategyLocalSecret)
+	kyma.Labels[v1beta2.SyncLabel] = v1beta2.EnableLabelValue
+	return kyma
+}
+
+func newKCPKymaWithNamespace(namePrefix, namespace, channel, syncStrategy string) *v1beta2.Kyma {
+	return builder.NewKymaBuilder().
+		WithNamePrefix(namePrefix).
+		WithNamespace(namespace).
+		WithAnnotation(watcher.DomainAnnotation, "example.domain.com").
+		WithAnnotation(v1beta2.SyncStrategyAnnotation, syncStrategy).
+		WithLabel(v1beta2.InstanceIDLabel, "test-instance").
+		WithChannel(channel).
+		Build()
+}
+
+func SyncKyma(ctx context.Context, clnt client.Client, kyma *v1beta2.Kyma) error {
+	err := clnt.Get(ctx, client.ObjectKey{
+		Name:      kyma.Name,
+		Namespace: kyma.Namespace,
+	}, kyma)
+	// It might happen in some test case, kyma get deleted, if you need to make sure Kyma should exist,
+	// write expected condition to check it specifically.
+	//nolint:wrapcheck
+	return client.IgnoreNotFound(err)
+}
+
+func KymaExists(ctx context.Context, clnt client.Client, name, namespace string) error {
+	kyma, err := GetKyma(ctx, clnt, name, namespace)
+	return CRExists(kyma, err)
+}
+
+func KymaDeleted(ctx context.Context,
```

Comment:
For these helper function can the return type just be bool? What we need to error for in assertions?

Reply:
Same for `KymaExists`

Reply:
The thing is return error can get more information for analysis if the test failed, but I will remove `ErrKymaNotDeleted`, instead just return `err`